aiohttp/tracing.py

```diff
@@ -24,31 +30,37 @@ class TraceConfig:
     """First-class used to trace requests launched via ClientSession
     objects."""
 
-    def __init__(self, trace_config_ctx_factory=SimpleNamespace):
-        self._on_request_start = Signal(self)
-        self._on_request_chunk_sent = Signal(self)
-        self._on_response_chunk_received = Signal(self)
-        self._on_request_end = Signal(self)
-        self._on_request_exception = Signal(self)
-        self._on_request_redirect = Signal(self)
-        self._on_connection_queued_start = Signal(self)
-        self._on_connection_queued_end = Signal(self)
-        self._on_connection_create_start = Signal(self)
-        self._on_connection_create_end = Signal(self)
-        self._on_connection_reuseconn = Signal(self)
-        self._on_dns_resolvehost_start = Signal(self)
-        self._on_dns_resolvehost_end = Signal(self)
-        self._on_dns_cache_hit = Signal(self)
-        self._on_dns_cache_miss = Signal(self)
-
-        self._trace_config_ctx_factory = trace_config_ctx_factory
-
-    def trace_config_ctx(self, trace_request_ctx=None):
+    def __init__(
+        self,
+        trace_config_ctx_factory: Type[SimpleNamespace]=SimpleNamespace
+    ) -> None:
+        self._on_request_start = Signal(self)  # type: Signal[Callable[['Application'], Awaitable[None]]] # noqa
```

Comment:
Maybe define `Signal[Callable[['Application'], Awaitable[None]]]` only once at top level and reuse it everywhere? You wouldn't need typing comments and qa disable in this case.

Reply:
Refactored